DONT MERGE: Primeng 18 with Lara Theme

fmontes opened the pull request:
# PrimeNG Components Modernization and UI Improvements

## Summary
This PR includes a comprehensive update to modernize our PrimeNG implementation, improve component accessibility, and clean up legacy code. The changes focus on updating deprecated modules, enhancing UI consistency, and improving the overall developer experience.

## Changes
- Updated TabView implementation to use modern `p-tabs` classes across components
- Replaced deprecated PrimeNG modules with their modern counterparts:
  - `InputTextareaModule` → `TextareaModule`
  - `DynamicDialogModule` → `DynamicDialog`
  - `ChipsModule` → `Chip` in autocomplete tags
- Enhanced form controls accessibility and layout:
  - Improved checkbox and radio button implementations
  - Removed redundant label prop from p-checkbox components
- Updated DotAlertConfirmComponent method implementations
- Implemented Lato font and refreshed UI styling
- Cleaned up cursor-related code for better maintainability
- 
## Additional Notes
These changes are part of our ongoing effort to modernize our UI components and maintain consistency with PrimeNG's latest best practices. The updates should improve both the developer experience and end-user accessibility.

fmontes committed:
Fix this

fmontes committed:
refactor(primeng): Replace deprecated InputTextareaModule with TextareaModule

fmontes committed:
refactor(primeng): Replace DynamicDialogModule with DynamicDialog

fmontes committed:
refactor(primeng): Remove label prop from p-checkbox

fmontes committed:
refactor(primeng): Improve checkbox and radio button accessibility and layout

fmontes committed:
refactor(primeng): Replace ChipsModule with Chip in autocomplete tags modules

fmontes committed:
refactor(primeng): Update DotAlertConfirmComponent method calls

fmontes committed:
clean up crazy cursor code

fmontes committed:
refactor(ui): Update UI components and styling with PrimeNG and Lato font

fmontes committed:
refactor(primeng): Update tabview classes to p-tabs in multiple components

zJaaal commented on core-web/apps/dotcms-ui/proxy-dev.conf.mjs:
```diff
@@ -19,7 +19,7 @@ export default [
             '/api/vtl',
             '/tinymce'
         ],
-        target: 'http://localhost:8080',
```
Beware of this

oidacra replied:
That is the URL of the proxy.

rjvelazco replied:
This is a dev config

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

nicobytes committed:
Merge branch 'main' into primeng-ui

freddyDOTCMS committed:
Fix: Record Content-Type of Binary Fields Correctly in Events (#31451)

The /dA syntax is: /dA/[content's id]/[binary file field variable name]
The /contentAsset syntax is: /contentAsset/image/[content's id]/[binary
file field variable name]

before the change we were just taking the Content's id

I am going to create a new issue for the /dotAsset/... url

### Proposed Changes
* No we are taking the content's id and the field name to look for the
right Content to be track


https://github.com/dotCMS/core/pull/31451/files#diff-9a756a0cf33f2fc28cfef619e51e6943ece96c251f3f7c98a77d3bb9e76b126aL88-R141

victoralfaro-dotcms committed:
#28422: Rebalancing ITs again part 2 (#31421)

valentinogiardino committed:
fix(FTM): avoid executing ftm logic for immediate publish date in live mode #31442 (#31448)

This pull request includes changes to the `PageResource` class and
updates to the Future Time Machine (FTM) tests. The most important
changes include importing additional utility classes, adding a new
method to handle the FTM grace window logic, and updating the test
scenarios to validate the new logic.

### Enhancements to Time Machine Functionality:

*
[`dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java`](diffhunk://#diff-225a703de3d295e7f6f361bd33455f27f09021c58403adbda9569399eef95da0R29):
Added `TimeMachineUtil` to handle time machine date parsing and
validation.
*
[`dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java`](diffhunk://#diff-d8f93700798e41883a5646a31a10a707d7832eaf17a8419196d5743eefd585e7R6-R23):
Introduced methods `parseTimeMachineDate` and `isOlderThanGraceWindow`
to validate and parse ISO 8601 date strings, ensuring dates are older
than a configurable grace window.

### Test Coverage Improvements:

*
[`dotcms-integration/src/test/java/com/dotcms/util/TimeMachineUtilTest.java`](diffhunk://#diff-9d38338bff66501965ab2d3ce3abad7acd7a88b962384e9e17e1a5898d6743a9R4-R28):
Added test cases for `parseTimeMachineDate` and `isOlderThanGraceWindow`
methods, covering scenarios with null, invalid, valid within grace
window, and valid outside grace window dates.
### Test Scenario Updates:

*
[`test-karate/src/test/java/graphql/ftm/setup.feature`](diffhunk://#diff-37f0b51013b66c2d7cf33fbd083df1c5ef5e71f805299725e33f0bd1de57b97cL41-R63):
Updated the setup to include a new content piece and its version within
the grace window. This ensures the test environment is correctly
configured for the new Time Machine scenarios.
*
[`test-karate/src/test/java/tests/graphql/ftm/CheckingTimeMachine.feature`](diffhunk://#diff-644e56e9ca92e212f8d0926f9788c1c35088eea4e1a6b2db75d6be1060523692R8-R10):
Added a new test scenario to verify that content with a publish date
within the grace window is not displayed by the Time Machine
functionality.

victoralfaro-dotcms committed:
#31452: removing fallback restore keys for release process (#31453)

Removing unnecessary restore keys property from cache action call.

fabrizzio-dotCMS committed:
fix(Page) Broken Assets Loading LIVE Mode  Refs: #31402 (#31455)

### Proposed Changes
* The Old TimeMachineFilter is blocking requests in Live Mode when the
tm_date is set in session
* I am deprecating this Filter in the hope it can be removed soon
* As for now I am disconnecting its execution with another session
variable to avoid breaking UX
* Also in this PR goes the fix that homologates the response in the
**urlContentMap** attribute

nicobytes committed:
refactor(data-access): Improve error message extraction in HTTP error manager (#31468)

### Issue Parent

#31384 

### Proposed Changes

This pull request focuses on improving the error handling in the
`DotHttpErrorManagerService` by updating the way `null` values are
handled in tests and by adding new methods for extracting error
messages. The most important changes include replacing `null` with
`undefined` in test cases and adding new private methods to handle error
message extraction.

### Error Handling Improvements:

* Updated test cases in `dot-http-error-manager.service.spec.ts` to
replace `null` with `undefined` in the `mockResponseView` function
calls. This change ensures consistency in how undefined values are
handled across the tests.
[[1]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL104-R104)
[[2]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL169-R169)
[[3]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL189-R189)
[[4]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL213-R213)
[[5]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL236-R236)
[[6]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL258-R258)
[[7]](diffhunk://#diff-06156a7fee1fe9027dab834d15ba8d8908e9a12555e462c05e28e38d4ef03abdL280-R280)

### Codebase Enhancements:

* Added a new private method `getErrorMessage` in
`dot-http-error-manager.service.ts` to extract a readable error message
from an `HttpErrorResponse`. This method improves the readability and
maintainability of the code by centralizing the error message extraction
logic.
* Introduced another private method `extractMessageFromErrorObject` to
handle different error object structures and trim messages if they
contain a colon. This method further refines the error message
extraction process.

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)


### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
<img width="1510" alt="Screenshot 2025-02-25 at 2 18 43 PM"
src="https://github.com/user-attachments/assets/dd1c0de1-b21f-415e-b256-5fba2ccdf393"
/> | <img width="1660" alt="Screenshot 2025-02-25 at 2 18 17 PM"
src="https://github.com/user-attachments/assets/91a1c034-da08-4b28-ad79-d318788bc4f5"
/>

KevinDavilaDotCMS committed:
fix(uve): Fixed iframeURL selector on store (#31470)

Traditional:


https://github.com/user-attachments/assets/57df3ca3-1536-46ba-bf4c-540898c53887


Headless:


https://github.com/user-attachments/assets/85fba492-e541-44ba-a95f-0c7894476aae

---------

Co-authored-by: Kevin Davila <56242609+kevindaviladev@users.noreply.github.com>

jcastro-dotcms committed:
feat(Content Analytics) #31319 : Add telemetry customer category, customer name, and environment name to Content Analytics events (#31459)

### Proposed Changes
* Adding four more attributes to a Customer Analytics Event:
  * Customer category.
  * Customer name.
  * Environment name.
  * Environment version.
* The schema and database table was updated in order to reflect this.
* The code changes were added to the `BasicProfileCollector` class. This
means that absolutely all Data Collectors will have access to these new
attributes, no matter where they're being triggered from.

dsilvam committed:
Script to report lead time to change for GH issues (#31479)

This scripts calculates and reports the time between issues are included
in a Sprint all the way until they get the "Customer Deployed" label.

jdotcms committed:
Issue 31420 datetool (#31447)

now the dateTool support toTimeStamp

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui

fmontes committed:
Merge branch 'main' into primeng-ui
